types/jest/index.d.ts

```diff
@@ -694,7 +694,7 @@ declare namespace jest {
          *   expect(mock).toBeCalledWith(expect.any(Number));
          * });
          */
-        any(classType: any): any;
+        any<T extends Constructor>(classType: T): T extends Func ? ReturnType<T> : InstanceType<T>;
```

Comment:
In the past the following snippet used to pass: `expect.any(BigInt)`. Following this change it does not pass anymore.

It fails with error:
> error TS2345: Argument of type 'BigIntConstructor' is not assignable to parameter of type 'Constructor'.

See:
![image](https://user-images.githubusercontent.com/5300235/200508426-8319bf8c-845e-445f-b9ce-9faf34d807ae.png)